### DISCUSSION TITLE
ktx2 texture memory

### DISCUSSION THREAD

**Discussion by August1996** [google/filament#5857] [Q&A]
Hello, I test some models with filament. There is no advantage with `ktx2`  in memory, which is differences with official introduction.

**official introduction**
![official introduction](https://user-images.githubusercontent.com/15972489/182147452-60a906d4-36b6-4bc8-aeb5-50cfce076598.png)

**Models**
one original model and two ktx2 models converted by `gltf-transform`

- suit.glb：original model
- suit_ktx2_etc1s.glb：converted by `gltf-transform` with `etc1s` format.
- suit_ktx2_uastc.glb：converted by `gltf-transform` with `uastc` format. 

**Test Result**
![Test Result](https://user-images.githubusercontent.com/15972489/182147921-9654750e-34b5-45c7-b2d6-5ba4de68f38f.png)

@prideout

**Comment by romainguy** [comment]
What memory are you measuring and how? Compressed textures will reduce GPU memory usage, not CPU memory usage.

**Reply by August1996** [reply]
> What memory are you measuring and how? Compressed textures will reduce GPU memory usage, not CPU memory usage.

I use [perfdog](https://perfdog.wetest.net/) to measue memory. The detail memory view show `gfx` and `native pass` memory usage, which are no much different between `png` and `ktx2`

<img width="1225" alt="image" src="https://user-images.githubusercontent.com/15972489/182186942-8bd6bbc3-8fef-458e-a6fe-1dd475056b7a.png">

**Comment by prideout** [comment]
Are you sure that "gfx" means "GPU only"?  I believe Android Studio can tell you GPU memory usage.  Your device also needs to support the appropriate extensions.

Side note: an additional benefit of KTX2 is that it avoids the overhead of decoding PNG or JPEG.

**Reply by August1996** [reply]
> Are you sure that "gfx" means "GPU only"? I believe Android Studio can tell you GPU memory usage. Your device also needs to support the appropriate extensions.
> 
> Side note: an additional benefit of KTX2 is that it avoids the overhead of decoding PNG or JPEG.

In fact, I've only looked at the total memory before.

I will try again with Android Studio.

**Reply by August1996** [reply]
> Are you sure that "gfx" means "GPU only"? I believe Android Studio can tell you GPU memory usage. Your device also needs to support the appropriate extensions.
> 
> Side note: an additional benefit of KTX2 is that it avoids the overhead of decoding PNG or JPEG.

I use Android Studio Profiler to test the memory, the conclusion is same to perfdog.

Here is the test result. @prideout 
[memory test.xlsx](https://github.com/google/filament/files/9239670/memory.test.xlsx)

**Reply by August1996** [reply]
> Are you sure that "gfx" means "GPU only"? I believe Android Studio can tell you GPU memory usage. Your device also needs to support the appropriate extensions.
> 
> Side note: an additional benefit of KTX2 is that it avoids the overhead of decoding PNG or JPEG.

When I render 7 models, the ktx2 texture does save memory than the image texture. I guess it is because the ktx2 decoder consumes a certain amount of memory, which makes the ktx2 memory advantage not obvious when I render a single model. @prideout
